### DISCUSSION TITLE
Does the battery icon on the Windows 10 taskbar not do anything now when clicked?

### DISCUSSION THREAD

**Discussion by DavidBerdik** [valinet/ExplorerPatcher#274] [Q&A] [answered]
I finally got around to installing ExplorerPatcher on my laptop today, and one thing that I have noticed is that the battery icon on the task bar does not seem to open the battery flyout when it is clicked. The battery icon still visually shows the battery life and when hovered on, the tooltip still works, but clicking on the battery does not seem to do anything.

I did not want to post this as a bug report because I am not sure if this is actually a bug or just something that is not possible to use with the 10 taskbar on 11.

**Comment by valinet** [comment, marked as answer]
Quoting from README:

> ### Fix the battery applet
>
> As you will notice, the battery flyout in the taskbar is broken in Windows 11. You can replace it with a much better alternative called [Battery Mode](https://en.bmode.tarcode.ru/) which has all the stock options and more.

It's broken, I haven't spent time attempting to find the cause and fix it because once I discovered that, which also let's me change external monitor brightness with ease (previously I had coded myself a separate utility for that), the Windows 10 battery flyout seems like a sad joke. This works, supports high DPI icons, shows messages in the system language etc. Properly done, so to speak, or at least I like it very much.

Thanks

**Reply by DavidBerdik** [reply]
Oh sorry about that! I guess I should learn to look at READMEs more carefully. This application looks quite promising. I will check it out!

**Reply by Anixx** [reply]
![изображение](https://user-images.githubusercontent.com/1271736/142149029-51bc4157-f9a2-4295-a685-4edd254bc720.png)
This is following the advice by @wixfigur.

**Reply by valinet** [reply]
Do you know what the DLL responsible for the battery flyout is, by any chance?

**Reply by valinet** [reply]
> 
> 
> ![изображение](https://user-images.githubusercontent.com/1271736/142149029-51bc4157-f9a2-4295-a685-4edd254bc720.png)
This is following the advice by @wixfigur.

Yeah, this has been working as well, but it's kind of obliterated by the Battery Mode app which does way more. I actually don't recommend this, as it does not support the Windows 10/11 modern power schemes (better battery, balanced, best performance) which are the power schemes which should actually be used nowadays instead of classic power schemes.

Also, to integrate it in EP is a bit more complicated, same strategy has to be done as with classic sound applet, which is to redirect the setting check for `UseWin32BatteryFlyout` in the battery flyout DLL from HKLM to HKCU so that the setting can be toggled from the Properties GUI. Otherwise, just use that registry entry and call it a day, not that big of a deal...

**Reply by wixfigur** [reply]
> 
> 
> Do you know what the DLL responsible for the battery flyout is, by any chance?

no, im sorry. 
getting back the windows 10 flyout would be nice. I like the performance slider in balanced mode very much. But in the end I dont use it much, switching between the power modes. If the power supply is plugged out, the system switches to saving battery. Thats all.

**Reply by valinet** [reply]
@wixfigur Implemented natively in the latest version:

![image](https://user-images.githubusercontent.com/6503598/142454619-7954e217-3965-4796-9540-00ad58a9f149.png)

**Reply by DavidBerdik** [reply]
@valinet Cool! I noticed that you have two battery icons in your task tray though. Is that a side effect of this feature?

![image](https://user-images.githubusercontent.com/7663225/142457026-10ba40f5-6285-4646-b0e8-50adcc4e2d42.png)

**Reply by valinet** [reply]
Does it say somewhere it's a side effect? It was inevitable such a question, I was almost afraid to push that screenshot. No, one is Windows' applet, one is Battery Mode.

**Reply by DavidBerdik** [reply]
@valinet Ah, sorry. I misunderstood.

**Reply by valinet** [reply]
Yeah, no worries. It works fine, I encourage you to check it out, it's easier to evaluate that way :D

**Reply by Anixx** [reply]
The new build is great!

**Reply by Anixx** [reply]
By the way, the Win8 Wi-Fi flyout does not work under Classic theme (this was the case on Win8 as well), so when not themed only the Win10 and Win11 options work well. I do not know if some kind of check or warning would be worth to implement.

**Reply by Anixx** [reply]
Also, given you implemented the LogonUI language flyout, I wonder, whether a LogonUI Wi-Fi flyout could be invoked.

**Reply by valinet** [reply]
> 
> 
> The new build is great!

Thanks <3

> 
> 
> By the way, the Win8 Wi-Fi flyout does not work under Classic theme (this was the case on Win8 as well), so when not themed only the Win10 and Win11 options work well. I do not know if some kind of check or warning would be worth to implement.

I don't think so, it is kind of expected. Not to mention, that is broken anyway, links do not work well, you cannot really connect to any WiFi etc. It's more of a show off feature, and I can't imagine who in this world prefers that, the layout is awful in my opinion. it's there for the sake of completeness.

> 
> 
> Also, given you implemented the LogonUI language flyout, I wonder, whether a LogonUI Wi-Fi flyout could be invoked.

I wrote a blog post about the technical implementation details: https://valinet.ro/2021/11/18/Functional-Windows-10-flyouts-on-Windows-11.html. tl;dr I tested that, it does not want to launch on the desktop, for some reason. Needs some more investigation which I am not going to do too soon, I am fine with the current implementation. It was my starting point though, as I noticed it and thought maybe I could enable that on the desktop as well. It's mostly a theme difference. There are a lot of modes the flyout can be toggled to, including some weird but familiar ones =)))) :

![image](https://user-images.githubusercontent.com/6503598/142466279-066793bf-4d81-486b-9115-ed4313b82bf1.png)

**Reply by DavidBerdik** [reply]
> Yeah, no worries. It works fine, I encourage you to check it out, it's easier to evaluate that way :D

I intend to. I just haven't had a chance yet and wanted to ask the question before it slipped my mind. 😁

> There are a lot of modes the flyout can be toggled to, including some weird but familiar ones

Wow. That looks weird. 😆

**Reply by justinlin099** [reply]
@valinet Looks like windows 10 battery flyout broke again in the newer version of windows, mine doesn't work

**Reply by valinet** [reply]
What’s the “newer version” of Windows 10? What build?

**Comment by wixfigur** [comment]
The classic battery flyout (as in windows 7) works. Maybe you can implement it in the settings. Its quite the same, as the suggested 3rd party battery manager.

The registry key is:
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\ImmersiveShell]
"UseWin32BatteryFlyout"=dword:00000001

**Reply by valinet** [reply]
Do you know what the DLL responsible for the battery flyout is, by any chance?

**Reply by DavidBerdik** [reply]
This is an interesting trick, but honestly, I think I would rather stick with [BatteryMode](https://github.com/tarcode-apps/BatteryMode). It strikes a good balance between being simple to use and advanced enough to support the additional features that were added to the Windows 10 battery flyout.

**Comment by jlearman** [comment]
The taskbar battery icon flyout for Win10 is not by any means a "sad joke."  It's the main reason I installed ExplorerPatcher.

On my Thinkpad, when my power plan is set to "Balanced", the flyout shows a "Power mode" slider with three settings: 
* better battery
* better performance
* best performance

I use this many times every day, as I like quiet operation most of the time (fan off), but I click to `best performance` when starting a build or running dozens of containers for testing.

When my power plan is set to anything other than "Balanced," the flyout isn't very useful, possibly leading to the "sad joke" assessment.

This feature is working perfectly for me, on version 22621.1555.55.1 .

**Reply by JTCozart** [reply]
I agree... If I am in balanced mode my CPU is limited to <=1GHz.. Switching this in settings gets annoying. I would very much like the battery icon to work again please. @valinet
